artifact/metadata.go

```diff
@@ -161,7 +161,11 @@ func (hi *HeaderInfo) GetArtifactProvides() *ArtifactProvides {
 }
 
 type HeaderInfoV3 struct {
-	Updates          []UpdateType      `json:"updates"`
+	// For historical reasons, "payloads" are often referred to as "updates"
+	// in the code, since this was the old name (and still is, in V2 and
+	// V1). This is the reason why the struct field is still called
+	// "Updates".
+	Updates          []UpdateType      `json:"payloads"`
```

Comment:
Should we grab this bull by the horns while we have the chance?

Reply:
I considered it, but the problem is that there are soooo many occurrences of the string "Update" used in various locations, often as part of longer identifiers, and in different contexts. So it's actually quite a lot of work to replace them, because search and replace won't work reliably. Since we are already on a tight schedule, I opted for changing only the ones that are user facing and the ones we cannot change later.

Reply:
FWIW, I also have a few followup renames in log messages that I missed the first time around.